### PR TITLE
CORE-4988 add metric to record the length of time it takes for a flow to run.

### PR DESCRIPTION
**Pull request by LWogan**
The current method of adding meters/tags is not ideal. Not sure if that should be changed in this PR or left to Dries and co.

### CODE REVIEW

File: libs/metrics/src/main/kotlin/net/corda/metrics/CordaMetrics.kt [hunk 1]
```diff
@@ -28,6 +28,11 @@ object CordaMetrics {
          * Time it took to create the sandbox
          */
         object SandboxCreateTime : Metric<Timer>("sandbox.create.time", Metrics::timer)
+
+        /**
+         * Time it took to create the sandbox
```
Comment (fowlerrr) [inline libs/metrics/src/main/kotlin/net/corda/metrics/CordaMetrics.kt]: I think you need to update this javadoc

File: components/flow/flow-service/src/main/kotlin/net/corda/flow/pipeline/handlers/requests/FlowFinishedRequestHandler.kt [hunk 2]
```diff
@@ -38,6 +39,7 @@ class FlowFinishedRequestHandler @Activate constructor(
     ): FlowEventContext<Any> {
         val checkpoint = context.checkpoint
         log.info("Flow [${checkpoint.flowId}] completed successfully")
+        recordFlowRuntimeMetric(checkpoint, "FlowFinished")
```
Comment (fowlerrr) [inline components/flow/flow-service/src/main/kotlin/net/corda/flow/pipeline/handlers/requests/FlowFinishedRequestHandler.kt]: Is there any way to determine the operational status without passing it in?  It'd just be a slightly nicer bit of code if you could.

I'd imagine making the record and extension on the checkpoint and just saying `checkpoint.recordFlowRuntimeMetric()`.

You can still make it an extension, of course, but the value decreased when you have to pass the string anyway.

Also, and maybe this isn't useful, but maybe make the strings for operational status constants? Or an enum.

Comment (fowlerrr) [reply components/flow/flow-service/src/main/kotlin/net/corda/flow/pipeline/handlers/requests/FlowFinishedRequestHandler.kt]: Having said that, maybe an extension isn't really great as it's not an operation _on the checkpoint_.  Still think the constants are nice, though

Comment (LWogan) [reply components/flow/flow-service/src/main/kotlin/net/corda/flow/pipeline/handlers/requests/FlowFinishedRequestHandler.kt]: >maybe this isn't useful, but maybe make the strings for operational status constants? Or an enum.

I've updated to use the FlowStates enum

**Review by fowlerrr — APPROVED** [review]

**Review by wzur-r3 — APPROVED** [review]
`build.gradle` LGTM

**Pull request merged**